From: Ming Lei <ming.lei@redhat.com>
Date: Wed, 27 Nov 2024 21:51:27 +0800
Subject: [PATCH] block: remove unnecessary check in blk_unfreeze_check_owner()
Git-commit: aff09dc1fd3a165289011ab23cc3b46978ec741c
Patch-mainline: v6.14-rc1
References: jsc#PED-9651

The following check of 'q->mq_freeze_owner != current' covers the
previous one, so remove the unnecessary check.

Signed-off-by: Ming Lei <ming.lei@redhat.com>
Link: https://lore.kernel.org/r/20241127135133.3952153-2-ming.lei@redhat.com
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 block/blk-mq.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 8ac19d4ae3c0..60f457f62913 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -142,8 +142,6 @@ static bool blk_freeze_set_owner(struct request_queue *q,
 /* verify the last unfreeze in owner context */
 static bool blk_unfreeze_check_owner(struct request_queue *q)
 {
-	if (!q->mq_freeze_owner)
-		return false;
 	if (q->mq_freeze_owner != current)
 		return false;
 	if (--q->mq_freeze_owner_depth == 0) {
-- 
2.35.3

